src/web_ui/src/context/auth0.context.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect, useContext, useReducer } from 'react';
+import createAuth0Client from '@auth0/auth0-spa-js';
+import LoginAction from "../store/actions/auth/LoginAction"
+import { useDispatch, useSelector} from "react-redux"
+
+const DEFAULT_REDIRECT_CALLBACK = () => window.history.replaceState({}, document.title, window.location.pathname);
+
+export const Auth0Context = React.createContext();
+
+export const useAuth0 = () => useContext(Auth0Context);
+
+export const Auth0Provider = ({ children, onRedirectCallback = DEFAULT_REDIRECT_CALLBACK, ...initOptions }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState();
+  const [user, setUser] = useState();
+  const [auth0Client, setAuth0] = useState();
+  const [loading, setLoading] = useState(true);
+  const [popupOpen, setPopupOpen] = useState(false);
+  const data = useSelector(state => state)
+  const dispatch = useDispatch()
+  
+  useEffect(() => {
+    const initAuth0 = async () => {
+      const auth0FromHook = await createAuth0Client(initOptions);
+      setAuth0(auth0FromHook);
+
+      if (window.location.search.includes('code=') && window.location.search.includes('state=')) {
+        const { appState } = await auth0FromHook.handleRedirectCallback();
+        onRedirectCallback(appState);
+      }
+
+      const isAuthenticated = {
+        isAuthenticated: await auth0FromHook.isAuthenticated(),
```

Comment:
expect isAuthenticated to exist and be updated by state, it can be a TODO for now